markets/perps-market/contracts/storage/PerpsMarket.sol

```diff
@@ -141,8 +143,19 @@ library PerpsMarket {
         Position.Data memory newPosition
     ) internal returns (MarketUpdateData memory) {
         Position.Data storage oldPosition = self.positions[accountId];
+
         int128 oldPositionSize = oldPosition.size;
 
+        // update the market debt correction accumulator before losing oldPosition details
+        // by adding the new updated notional (old - new size) plus old position pnl
+        // debtCorrectionAccumulator += newPosition.size * currentPrice.toInt() - oldPosition.size * currentPrice.toInt() + oldPositionPnl ;
+        uint currentPrice = newPosition.latestInteractionPrice;
+        (int oldPositionPnl, , , ) = oldPosition.getPnl(currentPrice);
+
+        self.debtCorrectionAccumulator +=
+            currentPrice.toInt().mulDecimal(newPosition.size - oldPositionSize) +
+            oldPositionPnl;
```

Comment:
do we need to do anything with accumulator on liquidiations?  i dont think so since it becomes realized right?  i'll think through this a bit more

Reply:
most likely yes. Maybe the numbers get balanced together, but not completely sure on some corner cases